backend/src/packages/articles/article.repository.ts

```diff
@@ -262,6 +319,32 @@ class ArticleRepository implements IArticleRepository {
       coverUrl: article.cover?.url ?? null,
     });
   }
+
+  public async toggleIsFavourite(
+    userId: number,
+    articleId: number,
+  ): Promise<number | FavouredUserArticlesModel | null> {
+    return await this.favouriteArticlesModel.transaction(async () => {
+      const currentData = await this.favouriteArticlesModel
+        .query()
+        .where({ userId })
+        .where({ articleId });
+
+      if (currentData.length === 0) {
+        return await this.favouriteArticlesModel
+          .query()
+          .returning(['article_id', 'user_id'])
+          .insert({ userId, articleId });
+      } else if (currentData.length === 1) {
+        return await this.favouriteArticlesModel
+          .query()
+          .delete()
+          .where({ articleId, userId });
+      }
+
+      return null;
```

Comment:
```suggestion
      if (!currentData) {
        return await this.favouriteArticlesModel
          .query()
          .insert({ userId, articleId })
          .returning(['article_id', 'user_id']);
      }
      
      return await this.favouriteArticlesModel
          .query()
          .delete()
          .where({ articleId, userId })
          .returning(['article_id', 'user_id']);
```

Reply:
Ah I see ,  after "first()" it's no longer an array.